app/src/main/java/com/abhijitvalluri/android/fitnotifications/services/NLService.java

```diff
@@ -443,16 +375,122 @@ private boolean appNotificationScheduleActive(String appPackageName) {
 
         int startTime = appSelection.getStartTime();
         int stopTime = appSelection.getStopTime();
-        Date currDate = new Date();
 
         // Get current time
         Calendar cal = Calendar.getInstance();
-        cal.setTime(currDate);
         int hour = cal.get(Calendar.HOUR_OF_DAY);
         int minute = cal.get(Calendar.MINUTE);
-        int currTime = hour*60 + minute;
+        int currTime = hour * 60 + minute;
+
+        return ((currTime >= startTime) && (currTime <= stopTime));
```

Comment:
I didn't mean to commit this, sorry.

By the way, checking for stopTime == 23:59 is not good enough. Imagine someone has a range of 6:00 a.m. - 5:59 a.m. Or someone else having a range of 8:00 a.m. - 11:59 p.m.

I think a better heuristic will be to include the last minute if it ends with '9'. I.e. stopTime == 18:39 will show notification sent at 18:39:07. But stopTime == 18:40 won't show notification at 18:40:21.
What do you think?

Otherwise you'd probably need that checkbox "Enable notifications schedule" in app preferences.